### PR TITLE
[Issue #68] Nginx Ingress Controller

### LINKED ISSUE #68: NGINX ingress

### PR DESCRIPTION
**Pull request by v-ctiutiu**
Fixes #68 

## Overview

- Added Nginx Ingress sub-chapter
- Adapted main chapter for both Ingresses: Ambassador and Nginx.
- Adapted guides (LB migration, wildcard certificates) to accommodate both Ingresses: Ambassador and Nginx.
- Added separate guide for Nginx performance tuning.
- Validated all Nginx steps from both guides and its dedicated chapter.
- Adapted links and overall layout.
- Created required diagrams, histograms, etc.

### CODE REVIEW

File: 03-setup-ingress-controller/guides/wildcard_certificates.md [hunk 1]
```diff
@@ -0,0 +1,682 @@
...
+
+  ```text
+  NAME                           READY   AGE
+  letsencrypt-ambassador-wcard   True    3m32s
+  ```
+
+- `Nginx` ingress:
+
+  ```shell
+  kubectl get issuer letsencrypt-nginx-wcard -n ingress-nginx
```
Comment (ChandanSagar) [inline 03-setup-ingress-controller/guides/wildcard_certificates.md:273]: The issuer is created in the backend namespace.
The command should be:
kubectl get issuer letsencrypt-nginx-wcard -n backend

File: 03-setup-ingress-controller/guides/wildcard_certificates.md [hunk 2]
```diff
@@ -0,0 +1,682 @@
...
+  Data
+  ====
+  tls.crt:  5632 bytes
+  tls.key:  1679 bytes
+  ```
+
+- `Nginx` ingress:
+
+  ```shell
+  kubectl describe secret starter-kit.online -n ingress-nginx
```
Comment (ChandanSagar) [inline 03-setup-ingress-controller/guides/wildcard_certificates.md:454]: The command should be:
kubectl describe secret starter-kit.online -n backend

File: 03-setup-ingress-controller/guides/wildcard_certificates.md [hunk 3]
```diff
@@ -0,0 +1,682 @@
...
+- `Ambassador Edge Stack` ingress:
+
+  ```shell
+  kubectl get certificate starter-kit.online -n ambassador
+  ```
+
+- `Nginx` ingress:
+
+  ```shell
+  kubectl get certificate starter-kit.online -n ingress-nginx
```
Comment (ChandanSagar) [inline 03-setup-ingress-controller/guides/wildcard_certificates.md:401]: The command should be:
kubectl get certificate starter-kit.online -n backend

**Review by ChandanSagar — CHANGES_REQUESTED** [review]

File: 03-setup-ingress-controller/guides/wildcard_certificates.md (hunk 1 above)
Comment (v-ctiutiu) [reply 03-setup-ingress-controller/guides/wildcard_certificates.md]: Should be fixed in the latest commit. Thanks.

File: 03-setup-ingress-controller/guides/wildcard_certificates.md (hunk 2 above)
Comment (v-ctiutiu) [reply 03-setup-ingress-controller/guides/wildcard_certificates.md]: Should be fixed in the latest commit. Thanks.

File: 03-setup-ingress-controller/guides/wildcard_certificates.md (hunk 3 above)
Comment (v-ctiutiu) [reply 03-setup-ingress-controller/guides/wildcard_certificates.md]: Should be fixed in the latest commit. Thanks.

**Review by ChandanSagar — APPROVED** [review]

**Review by KumariPurnima — APPROVED** [review]
Reviewed & Approved the changes.

**Pull request merged**